Лабораторная работа 2

WorkerStudy opened the pull request:
-Переход на экран с героем
-Загрузка изображений из интернета

WorkerStudy committed:
Лабораторная работа 2

-Переход на экран с героем
-Загрузка изображений из интернета

rvachev commented on Flutter/lib/Human.dart:
```diff
@@ -1,38 +1,66 @@
       name: "Юи Хирасава",
-      pathAvatar: "assets/img/ui.jpg",
-      info: "Not Info",
+      imageHero: Image.network(
+          "https://i.pinimg.com/736x/53/eb/23/53eb23e56606f4247c9e64730a5a9770.jpg",
+          errorBuilder:
+              (BuildContext context, Object exception, StackTrace? stackTrace) {
+        return Image.asset("assets/img/ui.jpg", fit: BoxFit.cover);
+      }, fit: BoxFit.cover),
```
хранить виджет в модели - нецелесобразно. Что, если карточка никогда не будет отрисована во время работы программы? Потратиться вычислительная мощность и время на все эти действия

rvachev commented on Flutter/lib/poster_hero.dart:
```diff
@@ -0,0 +1,53 @@
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
```
тут нереализована основная задача лабораторной - чтобы переход на экран с детальной информацией осуществлялся с анимацией, которую предоставляет виджет `Hero`

rvachev commented on Flutter/lib/poster_hero.dart:
```diff
@@ -0,0 +1,53 @@
+            child: Column(
+              crossAxisAlignment: CrossAxisAlignment.start,
+              children: [
+                Expanded(
+                  child: Container(),
+                ),
```
это выглядит как хак. Внутри `Stack` мы можем использовать виджет `Positioned` для позионирования виджетов относительно сторон

WorkerStudy committed:
Исправления

- реализована основная задача лабораторной работы 2
- больше никаких хаков
- теперь модель

rvachev reviewed (approved):
принято
